cmake/modules/AddCephTest.cmake

```diff
@@ -22,6 +22,22 @@ function(add_ceph_test test_name test_path)
   # none of the tests should take more than 1 hour to complete
   set_property(TEST ${test_name}
     PROPERTY TIMEOUT ${CEPH_TEST_TIMEOUT})
+  # crimson seastar unittest always run with --smp N
+  string(FIND "${ARGN}" "--smp" smp_pos)
+  if(NOT smp_pos EQUAL -1)
+    set(i 1)
+    while(i LESS ARGC)
+      list(GET ARGV ${i} argv_value)
+      math(EXPR i "${i} + 1")
+      if(argv_value STREQUAL "--smp")
+        list(GET ARGV ${i} smp_count)
+        break()
+      endif()
+    endwhile()
+    set(resource_groups "${smp_count},cpus:1")
+    set_property(TEST ${test_name}
+      PROPERTY RESOURCE_GROUPS ${resource_groups})
+  endif()
```

Comment:
this is not quite elegant. we could improve it in two ways.  now that we already know the index of `--smp` in the `${ARGN}`. why not just get its option with something like:
```cmake
if(smp_pos GREATER -1)
  if(smp_pos EQUAL ARGC) 
    message(FATAL_ERROR "${test_name} --smp requires an argument")
  endif()
  math(EXPR i "${i} + 1")
  list(GET ARGV ${i} smp_count)
  # we assign CPU cores as resources to each tests with "--smp" option.
  # each group represents a core of CPU, and each group has only a single
  # slot. here, the test in question requires:
  # ${smp_count} groups, each of which requires 1 slot.
  set_property(TEST ${test_name}
       PROPERTY RESOURCE_GROUPS "${smp_count},cpus:1")
endif()
```
or we can use  update the tests to pass smp and memory with keyword argument, and use `cmake_parse_arguments(). that'd be a more intrusive change.

so i'd suggest go with the first approach. the point is to add more comment to explain the concept of group and slot. and how they maps to the `--smp` option.

Reply:
@rosinL i still suggest to add some comment to explain the the way how we map the resources in CMake. as `RESOURCE_GROUPS` is not not well-known CMake technique, a comment should help other developers to under our project uses it.